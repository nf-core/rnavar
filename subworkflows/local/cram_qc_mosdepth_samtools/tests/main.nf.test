nextflow_workflow {

    name "Test Subworkflow CRAM_QC_MOSDEPTH_SAMTOOLS"
    script "../main.nf"
    workflow "CRAM_QC_MOSDEPTH_SAMTOOLS"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "subworkflows/cram_qc_mosdepth_samtools"
    tag "samtools/stats"
    tag "mosdepth"

    test("default") {
        when {

            workflow {
                """
                input[0] = Channel.of([
                    [id:'test'],
                    file(params.pipelines_testdata_base_path + '/genomics/homo_sapiens/illumina/bam/test.rna.paired_end.sorted.bam', checkIfExists: true),
                    file(params.pipelines_testdata_base_path + '/genomics/homo_sapiens/illumina/bam/test.rna.paired_end.sorted.bam.bai', checkIfExists: true)
                ])
                input[1] = Channel.value([
                    [id:'fasta'],
                    file(params.pipelines_testdata_base_path + '/genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                ])
                input[2] = Channel.value([
                    [id:'intervals'],
                    file(params.modules_testdata_base_path + '/genomics/homo_sapiens/genome/genome.bed', checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.out.collectEntries { key, val ->
                        if(key.matches("\\d+")) {
                            return null
                        }
                        return [key,val]
                    }.findAll { it != null }
                ).match() }
            )
        }
    }
}
