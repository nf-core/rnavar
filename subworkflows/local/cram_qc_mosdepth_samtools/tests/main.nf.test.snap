{
    "default": {
        "content": [
            {
                "reports": [
                    [
                        {
                            "id": "test"
                        },
                        "test.mosdepth.global.dist.txt:md5,da4f2a9da24de7763247444ed9fc6d53"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.mosdepth.region.dist.txt:md5,da4f2a9da24de7763247444ed9fc6d53"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.stats:md5,1d1697d8b45e9f1c3280020155ee23d5"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1ba7e2d7c7c5d0f310ceb2f8e8846a4a",
                    "versions.yml:md5,ebf112f4081a4bc0e13520fe22ac26bd"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-03-05T13:46:12.688489101"
    }
}